src/common/queries/useAccountBalance.ts

```diff
@@ -1,56 +1,45 @@
-import {
-  UseQueryOptions,
-  UseSuspenseQueryOptions,
-  useQuery,
-  useSuspenseQuery,
-} from '@tanstack/react-query';
+import { useQuery, useSuspenseQuery } from '@tanstack/react-query';
 
-import { AddressBalanceResponse } from '@stacks/stacks-blockchain-api-types';
-
-import { useApi } from '../api/useApi';
+import { getErrorMessage } from '../../api/getErrorMessage';
+import { useApiClient } from '../../api/useApiClient';
 import { ONE_MINUTE } from './query-stale-time';
 
 const ACCOUNT_BALANCE_QUERY_KEY = 'accountBalance';
 
-export function useAccountBalance(
-  address?: string,
-  options: Omit<UseQueryOptions<any, any, AddressBalanceResponse, any>, 'queryKey' | 'queryFn'> = {}
-) {
-  const api = useApi();
+export function useAccountBalance(address?: string) {
+  const apiClient = useApiClient();
   return useQuery({
     queryKey: [ACCOUNT_BALANCE_QUERY_KEY, address],
-    queryFn: () => {
+    queryFn: async () => {
       if (!address) return undefined;
-      const response = api.accountsApi.getAccountBalance({
-        principal: address,
+      const { data, error } = await apiClient.GET('/extended/v1/address/{principal}/balances', {
+        params: { path: { principal: address } },
       });
-      return response;
+      if (error) {
```

Comment:
Have we made sure all these errors are caught properly. I think it would be a great idea to use the logError util func that will log this error in sentry. I would probably add the aility to rethrow the error to that func. And perhaps we can create a util function to do all this when making these apiclient requests so that we don't need to do this in every api hook. This comment equally applies to all these hooks. Hopefully it will make it easier if I don't bog down your PR by repeating the same comment :)